dapps/marketplace/src/pages/App.js

```diff
@@ -60,12 +71,27 @@ class App extends Component {
     ) {
       this.setState({ displayMobileModal: true })
     }
+    if (
+      this.state.isBrave &&
+      this.state.displayBrowseModal === false &&
+      !this.state.browseModalDismissed
+    ) {
+      this.setState({ displayBrowseModal: true })
+    }
   }
 
   static getDerivedStateFromError(err) {
     return { hasError: true, err }
   }
 
+  async checkBrave() {
+    const isBrave = (await (await fetch(
+      'https://api.duckduckgo.com/?q=useragent&format=json'
```

Comment:
As opposed to relying on a yet another centralized r3rd party service, could we possibly do the Brave browser detection ourselves ?

I wonder for example if we could use the technique mentioned in this article https://www.ctrl.blog/entry/brave-user-agent-detection.html
It might yield some false positive on mobile so we would have to test a bit...

Reply:
I just checked the proposed code on that article and it doesn't detect brave (at least in my tests).
That's probably because of many versions of Brave ahead of that solution. 
Before using duckduckgo proposed solution, I did tests using the user agent variable, the number of loaded extensions, etc to try and detect brave consistently. It happens to be very hard to detect.
I'm sure we can implement a local detection for the current version of Brave if time is assigned to the task, I can do it if you will. But bear in mind that we will be merging a solution that may fail tomorrow or the next day, as soon as Brave release a new version (they usually change behaviors to keep blending with chrome).

I know that its kind of funky to rely on a 3rd party on this issue, but it seems duckduckgo keeps the Brave detection updated. So your call, reopen the issue if we prefer to implement a local detection, and i will look into it.

Reply:
Reading a bit about the Brave browser and its detection, it seems that this is a bit of a cat and mouse game. There are thing that have been working years/months ago and do not right now. Seems that brave is actively trying to fight pages detecting their browser. 

@franckc the solution from the article does not work for me either.

@MarioBlanco If you want to give it a go for a local solution I think this is a good place to start: https://github.com/brave/brave-browser/issues/5768 I have tested it and it was working for me on mobile/chrome/brave browser without false positives. I only haven't implemented the logic for detection of ad blocking. - have hardcoded it while testing.

If you want to finish it this is the branch where functionality except ad blocking is implemented: https://github.com/OriginProtocol/origin/pull/3177 There is also some cleanup how UI modals display.

Reply:
Ok, nice cleanup. Going to implement adblocking detection as you suggested.

Reply:
@MarioBlanco for now the brave modal stuff has been removed. From my testing experience the DApp works in Brave with default settings. And if you disable cookies it totally breaks. So the modal would never display. You can read more about it here:

https://github.com/OriginProtocol/origin/pull/3177

If you have a different experience, or think of a better solution than suggested in the linked PR please share it :)